### PR TITLE
Add Copied OrderListViewController and OrderListViewModel

### PR DESCRIPTION
**Pull request by shiki**
Part of the DiffableDataSource fix for #1543. This is a continuation of #2791.  

🇨🇦 Sorry for the 732 lines. The contents are mainly copy-pasta from the original `OrdersViewController` and `OrdersViewModel` code. 

## The Plan

The plan is still the same as what I mentioned in #2791. There will be **new** `OrderListViewController` and `OrderListViewModel` classes. These classes will be similar to the existing [`OrdersViewController`](https://github.com/woocommerce/woocommerce-ios/blob/develop/WooCommerce/Classes/ViewRelated/Orders/OrdersViewController.swift) and [`OrdersViewModel`](https://github.com/woocommerce/woocommerce-ios/blob/develop/WooCommerce/Classes/ViewRelated/Orders/OrdersViewModel.swift) classes except that the new versions will be using `DiffableDataSource` instead of `FetchedResultsController`. These new classes will only be used if the device is using iOS 13.0. Eventually, we will remove the old classes when the minimum is set to iOS 13.0.

Here's a diagram of the plan:

image

Please let me know if you have any concerns. 

## What This PR Does

This PR copies the original `OrdersViewController` and `OrdersViewModel` to the new classes. 

Source | Target 
--------|-------
`OrdersViewController`        |       `OrderListViewController`
`OrdersViewModel` | `OrderListViewModel`

The only changes from the originals are:

- I exluded the references to the [`OrdersViewModel.resultsController`](https://github.com/woocommerce/woocommerce-ios/blob/c720e7415d83df7412237d304d6d8fc3c16c8a35/WooCommerce/Classes/ViewRelated/Orders/OrdersViewModel.swift#L52-L76)
- I excluded the [`UITableViewDataSource` conformance](https://github.com/woocommerce/woocommerce-ios/blob/c720e7415d83df7412237d304d6d8fc3c16c8a35/WooCommerce/Classes/ViewRelated/Orders/OrdersViewController.swift#L455-L493). This will be replaced by `UITableViewDiffableDataSource`. 
- I [marked the classes](https://github.com/woocommerce/woocommerce-ios/blob/91b45a8477fbfab272a4788f1f57204beac4044c/WooCommerce/Classes/ViewRelated/Orders/OrderListViewController.swift#L27) as `iOS 13.0` only.
- I added WIP placeholders [like this](https://github.com/woocommerce/woocommerce-ios/blob/91b45a8477fbfab272a4788f1f57204beac4044c/WooCommerce/Classes/ViewRelated/Orders/OrderListViewController.swift#L344-L350) to make the code compile for now. These placeholders are where the `DiffableDataSource` implementation will be added. 

So, in summary, these classes do not do anything and do not use `DiffableDataSource` or the new [`FetchResultSnapshotsProvider`](https://github.com/woocommerce/woocommerce-ios/blob/issue/1543-add-empty-orders-vc/Yosemite/Yosemite/SnapshotsProvider/FetchResultSnapshotsProvider.swift) yet. I wanted to submit that as a **separate PR** to hopefully make it easier to review the actual implementation of `DiffableDataSource`. 

## Testing

These currently cannot be tested. Please make sure the build passes for now. 🙇 

## Submitter Checklist

- [x] If it's feasible, I have added unit tests. 
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Review by Ecarrion — APPROVED** [review]
Looks good, I'm really liking the diagrams 💪.

----

On another note, how do you feel on renaming `OrdersViewController` and `OrdersViewModel` to `LegacyOrdersViewController` and `LegacyOrdersViewModel`. This will help other members to know that there is a newer class in case someone has to work on them.

**Comment by shiki** [comment]
> On another note, how do you feel on renaming OrdersViewController and OrdersViewModel to LegacyOrdersViewController and LegacyOrdersViewModel. This will help other members to know that there is a newer class in case someone has to work on them.

That's an excellent idea, @Ecarrion! I created #2803

**Comment by shiki** [comment]
Thanks for reviewing, Ernesto!
